sympy/solvers/ode/single.py

```diff
@@ -791,66 +791,93 @@ def _get_general_solution(self, *, simplify: bool = True):
-    [Eq(Integral(1/sqrt(C1 + 2*Integral(g(_u), _u)), (_u, f(x))), C2 + x),
-    Eq(Integral(1/sqrt(C1 + 2*Integral(g(_u), _u)), (_u, f(x))), C2 - x)]
+    >>> eq = f(x).diff(x, x) - f(x).diff(x)**2
+    >>> dsolve(eq, hint='2nd_autonomous_nth')
+    Eq(f(x), log(C1/(C2 + x)))
 
-    >>> from sympy import exp, log
+    >>> from sympy import exp, log, cbrt
     >>> eq = f(x).diff(x, 2) - exp(f(x)) + log(f(x))
-    >>> dsolve(eq, simplify=False)
-    [Eq(Integral(1/sqrt(-2*_u*log(_u) + 2*_u + C1 + 2*exp(_u)), (_u, f(x))), C2 + x),
-    Eq(Integral(1/sqrt(-2*_u*log(_u) + 2*_u + C1 + 2*exp(_u)), (_u, f(x))), C2 - x)]
+    >>> dsolve(eq, hint='2nd_autonomous_nth', simplify=False)
+    [Eq(Integral(1/sqrt(-2*_v*log(_v) + 2*_v + C1 + 2*exp(_v)), (_v, f(x))), C2 + x),
+    Eq(Integral(1/sqrt(-2*_v*log(_v) + 2*_v + C1 + 2*exp(_v)), (_v, f(x))), C2 - x)]
+
+    >>> eq = f(x).diff(x, 2) + cbrt(f(x)) + 1/f(x)
+    >>> dsolve(eq, hint='2nd_autonomous_nth', simplify=False)
+    [Eq(sqrt(2)*Integral(1/sqrt(-3*_v**(4/3) + 2*C1 - 4*log(_v)), (_v, f(x))), C2 + x),
+    Eq(sqrt(2)*Integral(1/sqrt(-3*_v**(4/3) + 2*C1 - 4*log(_v)), (_v, f(x))), C2 - x)]
+
+    >>> eq = f(x).diff(x, x) + f(x).diff(x)**4
+    >>> dsolve(eq, hint='2nd_autonomous_nth', simplify=False)
+    [Eq((C1 + 2*f(x))**(3/2)/3, C2 + x), Eq((C1 + 2*f(x))**(3/2)/3, C2 - x)]
 
     References
     ==========
 
-    http://eqworld.ipmnet.ru/en/solutions/ode/ode0301.pdf
+    https://en.wikipedia.org/wiki/Autonomous_system_(mathematics)
     """
-    hint = "2nd_nonlinear_autonomous_conserved"
+
+    hint = "2nd_autonomous_nth"
     has_integral = True
     order = [2]
 
     def _wilds(self, f, x, order):
-        fy = Wild('fy', exclude=[0, f(x).diff(x), f(x).diff(x, 2)])
-        return (fy,)
+        a = Wild('a', exclude=[0, f(x).diff(x), f(x).diff(x, 2)])
+        n = Wild('n', exclude=[x, f(x), f(x).diff(x), f(x).diff(x, 2)])
+        return a, n
 
     def _equation(self, fx, x, order):
-        fy = self.wilds()[0]
-        return fx.diff(x, 2) + fy
-
-    def _verify(self, fx):
-        return self.ode_problem.is_autonomous
+        a, n = self.wilds()
+        return fx.diff(x, 2) + a*fx.diff(x)**n
 
     def _get_general_solution(self, *, simplify: bool = True):
-        g = self.wilds_match()[0]
+        a, n = self.wilds_match()
         fx = self.ode_problem.func
         x = self.ode_problem.sym
-        u = Dummy('u')
-        g = g.subs(fx, u)
-        C1, C2 = self.ode_problem.get_numbered_constants(num=2)
-        inside = -2*Integral(g, u) + C1
-        lhs = Integral(1/sqrt(inside), (u, fx))
-        return [Eq(lhs, C2 + x), Eq(lhs, C2 - x)]
-
+        (C1, C2, C3) = self.ode_problem.get_numbered_constants(num=3)
+        v = Dummy('v')
+        a = a.subs(fx, v)
+        if n == 2:
+            arg = exp(Integral(a, v))
+            farg = C1*Integral(arg, (v, fx))
+            return [Eq(farg, C2 + x)]
+        else:
+            C3 = n - 2
+            if n%2 != 0:
```

Comment:
What happens if `n` is a symbol?

Reply:
Should a union of various piecewise equations be returned if `n `is a `symbol` instead of a `number`?

Reply:
Is it possible to return a single expression that represents all cases?

Reply:
In which case, when `n` is a `symbol` or for both cases when `n` is `symbol` and `number`?

Reply:
Rather than checking whether or not `n` is a symbol or a number check the properties of `n` e.g. `if n.is_even`:
```python
In [1]: S(2).is_even
Out[1]: True

In [2]: S(3).is_even
Out[2]: False

In [3]: x = Symbol('x', even=True)

In [4]: x.is_even
Out[4]: True

In [5]: x.is_odd
Out[5]: False
```
That way `n` can be any arbitrary expression including something like `2*n + 1` where `n` is a an integer.

You need to be aware though that sometimes the result is `None` meaning unknown so there are always three cases to think about when checking these properties:
```python
In [6]: y = Symbol('y')

In [7]: print(y.is_even)
None
```
Does the method here require `n` to be an integer?

Reply:
> Does the method here require `n` to be an integer?
>
No, `n` can be any arbitrary parameter.